clang/lib/CodeGen/TargetInfo.cpp

```diff
@@ -9334,10 +9351,16 @@ void AMDGPUTargetCodeGenInfo::setTargetAttributes(
 
   const bool IsHIPKernel =
       M.getLangOpts().HIP && FD && FD->hasAttr<CUDAGlobalAttr>();
-
   if (IsHIPKernel)
     F->addFnAttr("uniform-work-group-size", "true");
 
+  const bool IsSYCLKernel =
+      FD && M.getLangOpts().SYCLIsDevice &&
+      F->getCallingConv() == llvm::CallingConv::AMDGPU_KERNEL;
```

Comment:
Is there a reason you can't use `SYCLKernelAttr` here?

Reply:
Good point, added in https://github.com/intel/llvm/pull/5855/commits/a3437b3f68167cb4d807b5173db41c29fae633f7